File: BEmu3/HistoricalDataRequest/RequestHistoricElementString.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BEmu3.HistoricalDataRequest
{
    public class RequestHistoricElementString : Element
    {
        private readonly string _elementName, _value;

        internal RequestHistoricElementString(string elementName, string value)
        {
            this._elementName = elementName;
            this._value = value;
        }

        public override int NumElements { get { return 0; } }
        public override int NumValues { get { return 1; } }
        public override Name Name { get { return new Name(this._elementName); } }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            string tabs = Types.IndentType.Indent(tabIndent);
            StringBuilder result = new StringBuilder();

            result.AppendFormat("{0}{1} = {2}{3}", tabs, this._elementName, this._value, Environment.NewLine);

            return result;
        }
    }
}

File: BEmu_cpp/src/HistoricalDataRequest/RequestHistoricElementString.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu" file="src/HistoricalDataRequest/RequestHistoricElementString.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "HistoricalDataRequest/RequestHistoricElementString.h"
#include "BloombergTypes/Name.h"
#include "Types/IndentType.h"

namespace BEmu
{
	namespace HistoricalDataRequest
	{
		RequestHistoricElementString::RequestHistoricElementString(const std::string& elementName, const std::string& value)
		{
			this->_elementName = elementName;
			this->_value = value;
		}

		RequestHistoricElementString::~RequestHistoricElementString()
		{
		}


		Name RequestHistoricElementString::name() const
		{
			Name result(this->_elementName.c_str());
			return result;
		}

		size_t RequestHistoricElementString::numValues() const
		{
			return 1;
		}

		size_t RequestHistoricElementString::numElements() const
		{
			return 0;
		}


		std::ostream& RequestHistoricElementString::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			std::string tabs = IndentType::Indent(level, spacesPerLevel);
			stream << tabs << this->_elementName << " = " << this->_value << std::endl;
			return stream;
		}

	}
}